doc/radosgw/metrics.rst

```diff
@@ -0,0 +1,331 @@
+
+    "rgw_op": [
+        {
+            "labels": {},
+            "counters": {
+                "put_ops": 2,
+                "put_b": 5327,
+                "put_initial_lat": {
+                    "avgcount": 2,
+                    "sum": 2.818064835,
+                    "avgtime": 1.409032417
+                },
+                "get_ops": 5,
+                "get_b": 5325,
+                "get_initial_lat": {
+                    "avgcount": 2,
+                    "sum": 0.003000069,
+                    "avgtime": 0.001500034
+                },
+                "del_obj_ops": 2,
+                "del_obj_bytes": 5327,
+                "del_obj_lat": {
+                    "avgcount": 2,
+                    "sum": 0.034000782,
+                    "avgtime": 0.017000391
+                },
+                "del_bucket_ops": 1,
+                "del_bucket_lat": {
+                    "avgcount": 1,
+                    "sum": 0.134003083,
+                    "avgtime": 0.134003083
+                },
+                "copy_obj_ops": 1,
+                "copy_obj_bytes": 5033,
+                "copy_obj_lat": {
+                    "avgcount": 1,
+                    "sum": 0.024000553,
+                    "avgtime": 0.024000553
+                },
+                "list_obj_ops": 1,
+                "list_obj_lat": {
+                    "avgcount": 1,
+                    "sum": 0.004000092,
+                    "avgtime": 0.004000092
+                },
+                "list_buckets_ops": 1,
+                "list_buckets_lat": {
+                    "avgcount": 1,
+                    "sum": 0.002300000,
+                    "avgtime": 0.002300000
+                }
+            }
+        },
+    ]
+
+User & Bucket Labels
+--------------------
+
+Op metrics can be labeled with user names or bucket names, enabling cluster admins to see op metrics by bucket or by user.::
+
+    "rgw_op": [
+        ...
+        {
+            "labels": {
+                "Bucket": "bucket1"
+            },
+            "counters": {
+                "put_ops": 2,
+                "put_b": 5327,
+                "put_initial_lat": {
+                    "avgcount": 2,
+                    "sum": 2.818064835,
+                    "avgtime": 1.409032417
+                },
+                "get_ops": 5,
+                "get_b": 5325,
+                "get_initial_lat": {
+                    "avgcount": 2,
+                    "sum": 0.003000069,
+                    "avgtime": 0.001500034
+                },
+                ...
+                "list_buckets_ops": 1,
+                "list_buckets_lat": {
+                    "avgcount": 1,
+                    "sum": 0.002300000,
+                    "avgtime": 0.002300000
+                }
+            }
+        },
+        ...
+    ]
+
+To enable op metrics to be labeled by user or bucket, a cache for the user or bucket counters must be enabled.
```

Comment:
In a large system with many users and buckets, it may not be tractable to export all metrics to Prometheus. For that reason, the collection of these labeled metrics is disabled by default.

Once enabled, the working set of tracked users and buckets is constrained to limit memory and database usage. As a result, the collection of these labeled metrics will not always be reliable.